### ISSUE TITLE
Add support for ed25519 keys

### ISSUE THREAD

**Issue by webbson** [skavanagh/KeyBox#223]
I would like it if support for ed25519 keys is added. I'm planning on migrating to them, but keybox is holding me back for now.

**Comment by skavanagh** [comment]
You can change the key types here

https://github.com/skavanagh/KeyBox/blob/5a7c534005cbd7e374d05c84a34e465b76ee7070/src/main/resources/KeyBoxConfig.properties#L8-L10

.. but I don't believe JSch supports ED25519 yet (ECDSA works)

**Comment by webbson** [comment]
Ah, is it possible to use two types? I only add my own keys which I've generated elsewhere.

**Comment by skavanagh** [comment]
The key generation settings are used for the key that is generated for KeyBox to connect to the servers and any key that is generated by users when managing keys. It only handles one type.

You can set forceUserKeyGeneration to 'false'
https://github.com/skavanagh/KeyBox/blob/master/src/main/resources/KeyBoxConfig.properties#L31-L32
and users would set the public key that they generate elsewhere (it should handle different types)

**Comment by webbson** [comment]
Okay. I´ve actually set it up with forceUserKeyGeneration to false. However when I paste an ed25519 key in the box it turns red and won´t let me save the key.

**Comment by skavanagh** [comment]
Yeah -It's using Jsch to validate the key format and since it's not a supported type it's tripping the validation. Should be in the following lines

https://github.com/skavanagh/KeyBox/blob/master/src/main/java/com/keybox/manage/action/AuthKeysAction.java#L314-L320

**Comment by qdrddr** [comment]
+1

**Comment by mostafabayat** [comment]
+1

**Comment by gitwittidbit** [comment]
+1
